pkg/terraform/gcp/terraform_gcp.go

```diff
@@ -0,0 +1,226 @@
+	networkRoot := "google_compute_network"
+	if networkExists {
+		networkRoot = "data.google_compute_network"
+	}
+	firewallBody.SetAttributeTraversal("network", hcl.Traversal{
+		hcl.TraverseRoot{
+			Name: networkRoot,
+		},
+		hcl.TraverseAttr{
+			Name: networkName,
+		},
+		hcl.TraverseAttr{
+			Name: "name",
+		},
+	})
+	firewallAllow := firewallBody.AppendNewBlock("allow", []string{})
+	firewallAllowBody := firewallAllow.Body()
+	firewallAllowBody.SetAttributeValue("protocol", cty.StringVal("tcp"))
+	allowPortList := []cty.Value{}
+	for i := range ports {
+		allowPortList = append(allowPortList, cty.StringVal(ports[i]))
+	}
+	firewallAllowBody.SetAttributeValue("ports", cty.ListVal(allowPortList))
+	var allowIPList []cty.Value
+	allowIPList = append(allowIPList, cty.StringVal(ipAddress))
+	firewallBody.SetAttributeValue("source_ranges", cty.ListVal(allowIPList))
+}
+
+// SetPublicIP attach static IP(s) to the associated Google VM instance(s)
+func SetPublicIP(rootBody *hclwrite.Body, nodeName string, numNodes uint32) {
+	staticIPName := fmt.Sprintf("static-ip-%s", nodeName)
+	eip := rootBody.AppendNewBlock("resource", []string{"google_compute_address", staticIPName})
+	eipBody := eip.Body()
+	eipBody.SetAttributeRaw("name", createCustomTokens(staticIPName))
+	eipBody.SetAttributeValue("count", cty.NumberIntVal(int64(numNodes)))
+	eipBody.SetAttributeValue("address_type", cty.StringVal("EXTERNAL"))
+	eipBody.SetAttributeValue("network_tier", cty.StringVal("PREMIUM"))
+}
+
+// createCustomTokens enables usage of ${} in terraform files
+func createCustomTokens(tokenName string) hclwrite.Tokens {
+	return hclwrite.Tokens{
+		{
+			Type:  hclsyntax.TokenOQuote,
+			Bytes: []byte(`"`),
+		},
+		{
+			Type:  hclsyntax.TokenQuotedLit,
+			Bytes: []byte(fmt.Sprintf(`%s-`, tokenName)),
+		},
+		{
+			Type:  hclsyntax.TokenTemplateInterp,
+			Bytes: []byte(`${`),
+		},
+		{
+			Type:  hclsyntax.TokenIdent,
+			Bytes: []byte(`count`),
+		},
+		{
+			Type:  hclsyntax.TokenDot,
+			Bytes: []byte(`.`),
+		},
+		{
+			Type:  hclsyntax.TokenIdent,
+			Bytes: []byte(`index`),
+		},
+		{
+			Type:  hclsyntax.TokenTemplateSeqEnd,
+			Bytes: []byte(`}`),
+		},
+		{
+			Type:  hclsyntax.TokenCQuote,
+			Bytes: []byte(`"`),
+		},
+	}
+}
+
+// SetupInstances adds google_compute_instance section in terraform state file where we configure all the necessary components of the desired GCE instance(s)
+func SetupInstances(rootBody *hclwrite.Body, networkName, sshPublicKey, ami, staticIPName, instanceName string, numNodes uint32, networkExists bool) {
+	gcpInstance := rootBody.AppendNewBlock("resource", []string{"google_compute_instance", "gcp-node"})
+	gcpInstanceBody := gcpInstance.Body()
+	gcpInstanceBody.SetAttributeRaw("name", createCustomTokens(instanceName))
+	gcpInstanceBody.SetAttributeValue("count", cty.NumberIntVal(int64(numNodes)))
+	gcpInstanceBody.SetAttributeValue("machine_type", cty.StringVal("e2-standard-8"))
```

Comment:
do we need issue to enable param of this?

Reply:
I mean adding a github issue so as in the future we support different machine types